.github/workflows/release-build.yml

```
@@ -1,50 +1,58 @@
 name: "Release Build"
 
 on:
+  workflow_dispatch:
   push:
-    tags:
-      - "v*"
+    branches:
+      - main
+      - dev
+    paths:
+      - ".github/workflows/release-build.yml"
+      - "android/**"
+      - "assets/**"
+      - "lib/**"
+      - ".releaserc.js"
+      - "pubspec.yaml"
 
 jobs:
   release:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Set env
-        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
           java-version: "17"
           distribution: "zulu"
+
       - uses: subosito/flutter-action@v2
         with:
           channel: "stable"
+
+      - name: Cache Node modules
+        uses: actions/cache@v3
```

Migrate deprecated Node version from 16 to 20

```suggestion
        uses: actions/cache@v4
```